hw/xbox/xid-sdl.c

```diff
@@ -194,36 +201,35 @@ static const XIDDesc desc_xid_xbox_gamepad = {
 
 #define BUTTON_MASK(button) (1 << ((button) - GAMEPAD_DPAD_UP))
 
-static void update_output(USBXIDState *s) {
-#ifdef FORCE_FEEDBACK
-    if (s->sdl_haptic == NULL) { return; }
+static SDL_HapticEffect haptic_effect_init(HapticEmulationMode mode, uint large_magnitude, uint small_magnitude) {
+    assert((large_magnitude <= XID_MAX_HAPTIC_VALUE) && (small_magnitude <= XID_MAX_HAPTIC_VALUE));
```

Comment:
Just use `uint16_t` for `_magnitude`? that way, you don't even need the `assert` - it implies the full range is used.